src/tests/JIT/Directed/Convert/out_of_range_fp_to_int_conversions.cpp

```diff
@@ -154,6 +154,9 @@ extern "C" DLLEXPORT  uint64_t ConvertDoubleToUInt64(double x, FPtoIntegerConver
                 return (uint64_t)ConvertDoubleToInt64(x - int64_max_plus_1, CONVERT_MANAGED_BACKWARD_COMPATIBLE_ARM32) + (0x8000000000000000);
             }
         }
+    
+    case CONVERT_MANAGED_BACKWARD_COMPATIBLE_X86_X64:
+        return ((input_val != input_val) || (input_val < 0) || (input_val >= uint64_max_plus_1)) ? UINT64_MAX : (uint64_t)input_val;
```

Comment:
This is separate because `x = trunc(x)` converts `-0.0001` to `-0.0` which becomes `0` on conversion, yes?

Reply:
Exactly. For the new instruction vcvtsd2usi, it converts -0.0001 to ulong.MaxValue but after truncation -0.0001 becomes -0 and is casted to 0. Hence it is separate.